src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1282,9 +1294,9 @@ static void addBsonFilter
   }
   else
   {
-    std::string details = std::string("unknown query operator: ") + opr;
+    std::string details = std::string("unknown query operator: '") + opr + "'";
     alarmMgr.badInput(clientIp, details);
-    pushBackFilter = false;
+    return false;
   }
 
   if (pushBackFilter)
```

Comment:
I only changed this part, I believe there are more instances of **pushBackFilter = false**.
I can fix this though if you want.